Listview migration completed

farfilli opened the pull request:
It needs extensive testing.

Todo:
- Review the message shown in TextBoxStatus, is it really needed to truncate the path?
- Review the way the list is refreshed. I noticed that the list is reloaded many times, which will reset the item processes icon; in my opinion, the list should be more static and reloaded only if the user presses a new button "reload". This way, we could add the function to drag & drop files in the list. We can discuss this.

rmcanany commented:
Looks awesome!  Can't wait to try it out!

rmcanany commented:
Love what you've done with the place!  Nice job.  I've done quite a bit of testing and haven't found a single bug.

I did have several comments/questions for you...

Listview control:

You're right, the file list is updating at least twice.  I found one source -- when one of the Files To Process radio buttons are changed, that causes two events to fire.  I tweaked it so it doesn't call ReconcileFormChanges() twice.

For a large number of files, clearing and repopulating the ListView takes a while.  I found two methods similar to SEApp.DelayCompute() -- ListView.BeginUpdate() and ListView.EndUpdate().  I added those in UpdateListViewFiles() and it seems to help.

If you run Housekeeper a second time (probably with different task selections) how hard would it be to reinitialize the checkmarks back to an unchecked state?  I'm thinking it could maybe happen in CheckStartConditions().

Also, it was a little confusing for me when I tried to select a couple of files to process.  I kinda expected the checkbox to become checked; I wonder if the icon should be blank (or come to think of it, maybe the file's type icon) until processing begins.

I believe my prompt "Select file(s) to process OR Select none to process all" needs to change.  This seems clunky: "Select files to process (ESC to unselect all) OR Select none to process all".  Do you have any ideas how could we say that more succinctly?  

Regarding the prompt, would it be possible to include it at the top of the Listview, above Name?  It might be more clear that it applies to the list.  If not, I might change the background color to match the Listview, move it down so they're touching, and stretch it to match the Listview width.

I agree that with the Listview, we no longer need to truncate the path.  That same function is in all the other task classes.  How about you fix them up like you did with PartTasks.vb?  I'll eventually remove all references to TruncateFullPath() and delete it.

One last thing on this topic.  It looks like TextBoxColumnWidth is no longer needed.  Is that right?  If so, I'll remove it.

Template text boxes and buttons:
On the Configuration Tab, the template text boxes and buttons were moved to the left.  Was that in preparation for something new, or can I stretch them back out?

Release notes:
I like to update the release notes while changes are still fresh in my mind.  Here's what I have so far:

User Interface

Made several changes to clarify and streamline user interactions.  (Thank you @Fiorini).  

-- Changed the file list to emphasize the file name, rather than its path.  Also, sorted the list according the file type.

-- Added icons to the file list showing each file's status during processing; if a lot of error icons pop up, you can stop processing and verify, for example, that the file selection is appropriate for the tasks being performed.

-- Made a cool new logo.  It appears on the Task Bar when Housekeeper is running; you can right-click it and pin it there for quick access.

-- Did a lot of cleanup on control location and spacing to make it quicker and easier to find settings on each Task Tab.

That's all for now...

farfilli commented:
We need to change the way the list it's handled.
I will create a branch to experiment with a new way to populate and handle the list.
The user will decide when reload and what to add and what to remove from the list.
We will have this functionality: Add by Folder (with subfolder optional), add by list (txt, csv, xls), add by drag & drop, delete selected items by del key, clear the whole list, filter by type, process all, process selected, save list (txt, csv, xls)

Next pull I have moved the TruncateFullPath into CommonTasks and deleted in other classes
Changed the prompt adding "(ESC to clean selections)"

TextBoxColumnWidth  can be deleted
Template textbox and buttons where moved accidentally
